src/pages/data_submission/v2/v2-models.ts

```diff
@@ -0,0 +1,258 @@
+import { Dataset, FileStorageObject } from 'src/types/model'
+
+export type StudyPropertyType = 'Boolean' | 'String' | 'Number' | 'Date' | 'Json'
+
+export class StudyProperty {
+  key: string
+  studyPropertyId?: number
+  studyId?: number
+  type: StudyPropertyType
+  value?: unknown
+
+  constructor(key: string, type: StudyPropertyType, value?: unknown, studyId?: number, studyPropertyId?: number) {
+    this.key = key
+    this.studyId = studyId
+    this.studyPropertyId = studyPropertyId
+    this.type = type
+    this.value = value
+  }
+
+  toJSON() {
+    const obj = {
+      key: this.key,
+      value: this.value,
+      studyId: this.studyId,
+      studyPropertyId: this.studyPropertyId,
+    }
+    if (!obj.studyId) {
+      delete obj.studyId
+    }
+    if (!obj.studyPropertyId) {
+      delete obj.studyPropertyId
+    }
+    return obj
+  }
+}
+
+export class StringStudyProperty extends StudyProperty {
```

Comment:
I think there are different approaches to implement this functionality -- I can imagine a more functional approach that returns a specific study property like `StringStudyProperty` instead of the class-based approach, but let's continue with this approach and see if it is an effective mechanism for type safety, extensibility, and refactoring older components.